airflow/models/xcom.py

```diff
@@ -68,12 +80,22 @@ def __repr__(self):
 
     @classmethod
     @provide_session
-    def set(cls, key, value, execution_date, task_id, dag_id, session=None):
+    def set(cls, key, value, task_id, dag_id, execution_date=None, run_id=None, session=None):
         """
         Store an XCom value.
 
         :return: None
         """
+        if not (execution_date is None) ^ (run_id is None):
+            raise ValueError("Exactly one of execution_date or run_id must be passed")
+
+        if run_id:
+            from airflow.models.dagrun import DagRun
+
+            dag_run = session.query(DagRun).filter_by(dag_id=dag_id, run_id=run_id).one()
+
+            execution_date = dag_run.execution_date
+
         session.expunge_all()
```

Comment:
😞

Reply:
Yuuuup. I wasn't touching it on purpose in this PR.